plugin/kubernetes/setup.go

```diff
@@ -104,6 +104,17 @@ func kubernetesParse(c *caddy.Controller) (*Kubernetes, dnsControlOpts, error) {
 
 		for c.NextBlock() {
 			switch c.Val() {
+			case "endpoints":
+				args := c.RemainingArgs()
```

Comment:
I would reverse this and not use a switch:
~~~
if len(args) != 1 { 
     return nil, opts. c.ArgErr()
}
if args[0] != endpointNameModeName {
    return nil, opts. fmt.Errorf("wrong value for endpoints: %s")
}